tests/subscription/test_satellite.py

```diff
@@ -1,6 +1,530 @@
+        satellite.activation_key_create(activation_key)
+        satellite.activation_key_update(auto_attach='yes')
+
+        sm_guest_ack.unregister()
+        sm_guest_ack.register()
+        consumed_data = sm_guest_ack.consumed(vdc_virtual_sku)
+        assert consumed_data['temporary'] is True
+
+        _ = virtwho.run_cli()
+        sm_guest_ack.refresh()
+        consumed_data = sm_guest_ack.consumed(vdc_virtual_sku)
+        assert consumed_data['temporary'] is False
+
+    def test_vdc_virtual_subscription_on_webui(
+            self, virtwho, sm_guest, satellite, hypervisor_data,
+            vdc_pool_physical
+    ):
+        """Test the virtual vdc content subscriptions on satellite webui
+
+        :title: virt-who: satellite: test vdc virtual subscriptions on webui
+        :id: 5c71d37c-b5e2-4789-87f6-24357a96819b
+        :caseimportance: Medium
+        :tags: tier2
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Register guest to entitlement server
+            2. Attach physical vdc pool to hypervisor
+            3. Check the virtual vdc subscription
+            4. Attach virtual vdc pool to sm
+            5. Check the virtual vdc subscription again
+
+        :expectedresults:
+
+            1. When guest doesn't attach any virtual vdc, the subscription
+                consumed number is 0
+            2. When guest attached virtual vdc, the subscription consumed number
+                will not be 0
+        """
+        hypervisor_hostname = hypervisor_data['hypervisor_hostname']
+        if HYPERVISOR == 'libvirt-local':
+            hypervisor_name = [f'{hypervisor_hostname}']
+        else:
+            key = f'virt-who-{hypervisor_hostname}'
+            hypervisor_name = [key, key.lower()]
+
+        # check the subscription without virtual vdc pool attached for guest
+        satellite.host_delete(host=hypervisor_hostname)
+        _ = virtwho.run_cli()
+        satellite.attach(host=hypervisor_hostname, pool=vdc_pool_physical)
+        vdc_virt_data = sm_guest.available(vdc_virtual_sku, 'Virtual')
+        vdc_virt_pool = vdc_virt_data['pool_id']
+        subscription = satellite.subscription_on_webui(vdc_virt_pool)
+        assert (subscription['type'] == 'STACK_DERIVED'
+                and subscription["virt_only"] is True
+                and any(key in subscription["hypervisor"]["name"]
+                        for key in hypervisor_name))
+        assert (subscription["available"] == -1
+                and subscription["quantity"] == -1
+                and subscription["consumed"] == 0)
+
+        # check the subscription without virtual vdc pool attached for guest
+        sm_guest.refresh()
+        sm_guest.attach(pool=vdc_virt_pool)
+        for i in range(3):
+            time.sleep(30)
+            subscription = satellite.subscription_on_webui(vdc_virt_pool)
+            assert subscription["consumed"] == 1
+
+    # def test_register_by_item_on_webui(
```

Comment:
it seems the register by has not been supported any more, and in the old case, we also made workaround to pass it.